# In the NBA, is there an effect of being a forward on your salary?

In [11]:
install.packages("wooldridge")
library(wooldridge)
library(ggplot2)
library(tidyverse)

Installing package into ‘/srv/r’
(as ‘lib’ is unspecified)



In [12]:
nbasal

marr wage      exper age coll games minutes guard forward center ⋯ assists
1   1    1002.5     4    27  4    77    2867    1     0       0      ⋯  4.5   
2   1    2030.0     5    28  4    78    2789    1     0       0      ⋯  8.8   
3   0     650.0     1    25  4    74    1149    0     0       1      ⋯  0.2   
4   0    2030.0     5    28  4    47    1178    0     1       0      ⋯  1.5   
5   0     755.0     3    24  4    82    2096    1     0       0      ⋯  2.6   
6   0    2014.5     9    31  4    82    1971    0     1       0      ⋯  1.5   
7   1    1065.0     1    28  0    80    2303    0     1       0      ⋯  1.4   
8   0     420.0     3    27  3    67    1131    0     1       0      ⋯  0.7   
9   1     150.0     1    25  4    60     542    1     0       0      ⋯  2.0   
10  1    3050.0    12    35  3    74    2700    0     1       0      ⋯  2.3   
11  1    1030.9     9    31  4    70    2337    1     0       0      ⋯  6.9   
12  0     182.0     2    25  4    74     983    1     0       0      ⋯  2.2   
13  1     900.0     7    29  4    77    2746    1     0       0      ⋯ 10.1   
14  0     850.0     1    23  4    51     767    0     1       0      ⋯  1.2   
15  1    1004.0     8    30  4    82    2173    1     0       0      ⋯  2.7   
16  1    2040.0     6    28  4    82    2648    1     0       0      ⋯  2.6   
17  1    3125.0     3    25  4    51    1757    0     1       0      ⋯  3.6   
18  0    3120.0     2    24  4    60    2018    0     0       1      ⋯  1.4   
19  0    4000.0    18    41  4    74    1987    0     0       1      ⋯  1.1   
20  0     570.0     8    30  4    50    1005    1     0       0      ⋯  2.1   
21  0     620.0     6    27  4    82    2770    1     0       0      ⋯  4.0   
22  0     150.0     3    25  4    36     419    1     0       0      ⋯  1.0   
23  1    4000.0     8    31  0    75    2856    1     0       0      ⋯  4.6   
24  0     150.0     8    30  4    34     682    0     1       0      ⋯  1.0   
25  1    1007.5     1    26  0    75    1808    0     1       0      ⋯  3.4   
26  1    1059.0     4    26  4    76    1502    0     0       1      ⋯  1.4   
27  1    1020.0     7    29  4    43     397    0     0       1      ⋯  0.8   
28  1    3007.5     8    29  4    72    2759    1     0       0      ⋯  5.6   
29  1     150.0     8    31  4    76    1371    0     0       1      ⋯  0.9   
30  1    1120.0     9    31  4    51     814    1     0       0      ⋯  1.6   
⋮   ⋮    ⋮         ⋮     ⋮   ⋮    ⋮     ⋮       ⋮     ⋮       ⋮      ⋱ ⋮      
240 0     150.0000  1    24  4    75    1321    1     0       0      ⋯ 1.0    
241 1    2004.0000  6    27  4    60    2025    1     0       0      ⋯ 3.1    
242 0    2000.0000  1    23  4    65    1604    1     0       0      ⋯ 1.9    
243 0    2312.0000  8    30  4    69    1485    0     0       1      ⋯ 0.8    
244 1    1650.0000  2    25  4    78    2795    0     1       0      ⋯ 3.5    
245 0     601.0000  2    24  4    75    2487    0     1       0      ⋯ 2.1    
246 0     500.0000  1    23  0    54     650    0     0       1      ⋯ 0.3    
247 1     500.0000  2    25  4    61     749    1     0       0      ⋯ 1.5    
248 1     283.0000  2    26  4    65    1035    1     0       0      ⋯ 3.3    
249 0    1722.0000  8    30  4    82    2303    1     0       0      ⋯ 6.1    
250 1     600.0000  3    26  4     3      35    0     1       0      ⋯ 0.7    
251 1     400.0000  6    30  4    73    1397    0     1       0      ⋯ 0.5    
252 0    1400.0000  3    26  4    82    2605    1     0       0      ⋯ 2.3    
253 1    1630.0000  5    27  4    81    2915    1     0       0      ⋯ 9.7    
254 1    1400.0000  9    31  4    82    2149    0     1       0      ⋯ 1.5    
255 0     650.0000  1    23  4    16     107    0     1       0      ⋯ 0.5    
256 1    1200.0000 11    33  4    82    2147    1     0       0      ⋯ 3.3    
257 0    1228.0000  6    28  4    82    1608    0     0       1      ⋯ 0.7    
258 0    1510.0000  7    30  3    82    2539    0     

In [13]:
summary(nbasal)

      marr             wage          exper             age       
 Min.   :0.0000   Min.   : 150   Min.   : 1.000   Min.   :21.00  
 1st Qu.:0.0000   1st Qu.: 650   1st Qu.: 2.000   1st Qu.:25.00  
 Median :0.0000   Median :1186   Median : 4.000   Median :27.00  
 Mean   :0.4424   Mean   :1424   Mean   : 5.119   Mean   :27.39  
 3rd Qu.:1.0000   3rd Qu.:2014   3rd Qu.: 7.000   3rd Qu.:30.00  
 Max.   :1.0000   Max.   :5740   Max.   :18.000   Max.   :41.00  
                                                                 
      coll           games          minutes         guard       
 Min.   :0.000   Min.   : 3.00   Min.   :  33   Min.   :0.0000  
 1st Qu.:4.000   1st Qu.:57.00   1st Qu.: 983   1st Qu.:0.0000  
 Median :4.000   Median :74.00   Median :1690   Median :0.0000  
 Mean   :3.717   Mean   :65.72   Mean   :1682   Mean   :0.4201  
 3rd Qu.:4.000   3rd Qu.:79.00   3rd Qu.:2438   3rd Qu.:1.0000  
 Max.   :4.000   Max.   :82.00   Max.   :3533   Max.   :1.0000  
                 

In [14]:
model <- lm(lwage ~ forward, data = nbasal)

summary(model)


Call:
lm(formula = lwage ~ forward, data = nbasal)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.04543 -0.40353  0.08439  0.72040  1.77471 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  6.88051    0.06969  98.729   <2e-16 ***
forward      0.17555    0.10898   1.611    0.108    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.8788 on 267 degrees of freedom
Multiple R-squared:  0.009625,	Adjusted R-squared:  0.005916 
F-statistic: 2.595 on 1 and 267 DF,  p-value: 0.1084


## We can see here that being a forward increases wage by approximatelty 17.56% and this is not statistically significant because it is less than two times the standard error.

## However, we believe there are a lot of confounding variables so let's take this opprtunity to measure for omitted variable bias with a few variables

## 1. exper - how many years the player has been a professional
## 2. allstar - binary variable of whether the player has been an allstar or not
## 3. games - average amount of games a player plays in a year

## __________________________________________________________________________________________________________________________________________
## Let's start with exper and run the long regression

In [15]:
model_2 <- lm(lwage ~ forward + exper, data = nbasal)

summary(model_2)


Call:
lm(formula = lwage ~ forward + exper, data = nbasal)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.1210 -0.3374  0.1011  0.5284  1.7873 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  6.34139    0.09761  64.968  < 2e-16 ***
forward      0.17569    0.09968   1.763   0.0791 .  
exper        0.10531    0.01444   7.293 3.48e-12 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.8037 on 266 degrees of freedom
Multiple R-squared:  0.1747,	Adjusted R-squared:  0.1684 
F-statistic: 28.14 on 2 and 266 DF,  p-value: 8.178e-12


## Here we see that including experience actually increases the impact of being a forward on your salary and the t value increases. However it is still statistically insignificant.

In [16]:
model_3 <- lm(exper ~ forward, data = nbasal)

summary(model_3)


Call:
lm(formula = exper ~ forward, data = nbasal)

Residuals:
   Min     1Q Median     3Q    Max 
-4.120 -3.118 -1.118  1.882 12.880 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  5.119497   0.270147  18.951   <2e-16 ***
forward     -0.001315   0.422454  -0.003    0.998    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 3.406 on 267 degrees of freedom
Multiple R-squared:  3.629e-08,	Adjusted R-squared:  -0.003745 
F-statistic: 9.69e-06 on 1 and 267 DF,  p-value: 0.9975


## It looks like being a forward decreases your amount of experience however this is not statistically significant.

## Let's caclualte OVB

In [17]:
ovb_exper <- -0.001315 * 0.10531
ovb_exper

[1] -0.0001384826

## We do not gain much from this regression. There is a lot of stastistical insignificance so it is hard to make a conclusion. Let's check out the impact of being an allstar on your salary

## Again let's start out with the long regression

In [18]:
model_4 <- lm(lwage ~ forward + allstar, data = nbasal)

summary(model_4)


Call:
lm(formula = lwage ~ forward + allstar, data = nbasal)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.5905 -0.3858  0.1275  0.5945  1.5440 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  6.78271    0.06934  97.813  < 2e-16 ***
forward      0.18407    0.10422   1.766   0.0785 .  
allstar      0.81840    0.16047   5.100 6.46e-07 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.8403 on 266 degrees of freedom
Multiple R-squared:  0.09784,	Adjusted R-squared:  0.09106 
F-statistic: 14.42 on 2 and 266 DF,  p-value: 1.129e-06


## Here, forward's impact on salary has increased however still statistically insignifcant. What stands out here is that being an allstar increases your salary by approximately 81.8% and this is statistically significant.

## Now let's run the auxiliary regression

In [19]:
model_5 <- lm(allstar ~ forward, data = nbasal)

summary(model_5)


Call:
lm(formula = allstar ~ forward, data = nbasal)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.1195 -0.1195 -0.1195 -0.1091  0.8909 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  0.11950    0.02541   4.702 4.14e-06 ***
forward     -0.01041    0.03974  -0.262    0.794    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.3205 on 267 degrees of freedom
Multiple R-squared:  0.0002567,	Adjusted R-squared:  -0.003488 
F-statistic: 0.06855 on 1 and 267 DF,  p-value: 0.7937


## Again there is no statistical significance.

## Let's try out games

In [20]:
model_6 <- lm(lwage ~ forward + games, data = nbasal)

summary(model_6)


Call:
lm(formula = lwage ~ forward + games, data = nbasal)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.13782 -0.46908  0.08816  0.56538  1.95643 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 5.754140   0.185198  31.070  < 2e-16 ***
forward     0.163307   0.101460   1.610    0.109    
games       0.017214   0.002651   6.493  4.1e-10 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.818 on 266 degrees of freedom
Multiple R-squared:  0.1451,	Adjusted R-squared:  0.1387 
F-statistic: 22.58 on 2 and 266 DF,  p-value: 8.764e-10


## Again no stat sig